From f4e61504a457395018c02bafcf17d1e3f8644b78 Mon Sep 17 00:00:00 2001
From: Karel Zak <kzak@redhat.com>
Date: Fri, 9 Dec 2016 09:47:33 +0100
Subject: [PATCH] hwclock: don't check for permissions

The right place to verify user permissions is kernel. The current
situation is too strict (due to previous attempts to use hwclock as
suid that has never been supported).

Signed-off-by: Karel Zak <kzak@redhat.com>

Upstream-Status: Backport [f4e61504a457 on branch origin/master]

---
 sys-utils/hwclock.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/sys-utils/hwclock.c b/sys-utils/hwclock.c
index 21caeb2..d36c9c9 100644
--- a/sys-utils/hwclock.c
+++ b/sys-utils/hwclock.c
@@ -1859,11 +1859,6 @@ int main(int argc, char **argv)
 	argc -= optind;
 	argv += optind;
 
-	if (getuid() != 0) {
-		warnx(_("Sorry, only the superuser can use the Hardware Clock."));
-		hwclock_exit(EX_NOPERM);
-	}
-
 #ifdef HAVE_LIBAUDIT
 	if (testing != TRUE) {
 		if (adjust == TRUE || hctosys == TRUE || systohc == TRUE ||
-- 
2.10.2

